sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -152,5 +162,36 @@ private int onConnectionEvent(final RntbdConnectionEvent event, final Throwable
 
         return 0;
     }
+
+    public void attemptBackgroundAddressRefresh(RxDocumentServiceRequest request) {
+
+        if (request.requestContext == null) {
+            return;
+        }
+
+        if (request.requestContext.isRequestCancelledOnTimeout() == null) {
+            return;
+        }
+
+        AtomicBoolean isRequestCancelledOnTimeout = request.requestContext.isRequestCancelledOnTimeout();
+        final boolean forceAddressRefresh = request.requestContext.forceRefreshAddressCache;
+
+        if (this.forceBackgroundAddressRefreshInProgress.compareAndSet(false, true)
```

Comment:
should we also check whether for this request + exception, refresh address is needed?

Reply:
Sure - I'll scope it down to a few selected `CosmosException` sub-types. I had it this way to be more generic and since we only know where to call this method (acts as a way of scoping).